### PR TITLE
PSQLODBC Datatype Test for TIME

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Description

Adds ODBC Breakthrough Datatype Test for `TIME`
- Create Table
- Insert valid values
- Insert invalid values
- Update table with valid values
- Update table with invalid values
- Comparison Operators
- Table View
- Unique Constraint
- Composite Key

Task: BABELFISH-577
Signed-off-by: Colin Yuen [yuenhcol@amazon.com](mailto:yuenhcol@amazon.com)

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/psqlodbc/time.cpp [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+    // Below are valid in SQL Server
+    "00:00:00.00 AM"
+    "00AM"
+    "00 AM"
+  };
+
+  createTable(ServerType::MSSQL, BBF_TABLE_NAME, TABLE_COLUMNS);
+
+  testInsertionFailure(ServerType::MSSQL, BBF_TABLE_NAME, COL1_NAME, INVALID_INSERTED_VALUES, false);
+  // testInsertionFailure(ServerType::PSQL, PG_TABLE_NAME, COL1_NAME, INVALID_INSERTED_VALUES, false);
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/time.cpp]: Is there a reason that this is commented out or should it be deleted? If there is a reason it is commented out, please explain why.

File: test/odbc/psqlodbc/time.cpp [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+
+  const vector<string> UPDATED_VALUES = {
+    "12:34:56",
+    // "23:59:59.1",
+    // "23:59:59.22",
+    // "23:59:59.333",
+    // "23:59:59.4444",
+    // "23:59:59.55555",
+    // "23:59:59.666666",
+    "00:00:00"
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/time.cpp:198]: Could we test updating the value to NULL?

File: test/odbc/psqlodbc/time.cpp (hunk 1 above)
Comment (ColinKYuen) [reply test/odbc/psqlodbc/time.cpp]: Good catch, this shouldn't be commented out. Will fix.

File: test/odbc/psqlodbc/time.cpp (hunk 2 above)
Comment (ColinKYuen) [reply test/odbc/psqlodbc/time.cpp:198]: Will have to wait until #718 gets merged in as changes to support `NULL` for updates are in there.

Comment (ColinKYuen) [reply test/odbc/psqlodbc/time.cpp:198]: Done.

**Review by faviansamatha — APPROVED** [review]

**Review by rossd-bitquill — APPROVED** [review]

**Pull request merged**